crowbar_framework/app/models/service_object.rb

```diff
@@ -1852,4 +1852,41 @@ def node_changed_roles?(node, old_role, new_role)
     roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
     roles_in_old != roles_in_new
   end
+
+  # return true if no other attribute than the ignored has changed. And that the node has no other
+  # role than the roles in only_for_roles
+  def relevant_attributes_changed_if_roles?(node, old_role, new_role, ignore_attr, only_for_roles)
+    # get roles for this node
+    roles_in_new = new_role.elements.keys.select { |r| new_role.elements[r].include?(node) }.sort
+
+    # return true if node has other roles that ones in the list (only_for_roles)
+    return true unless Set.new(roles_in_new).subset?(Set.new(only_for_roles))
+
+    # prepare a clone of default attributes of old and new roles
+    old_selected_attributes = old_role.default_attributes[@bc_name].clone
+    new_selected_attributes = new_role.default_attributes[@bc_name].clone
+
+    # function to remove all ignored attributes from a list
+    remove_ignored = lamdba do |list, ignored|
+      ignored.each do |path|
+        iterator = list
+        path = path.split(".")
+
+        while path.length > 1
+          iterator = iterator[path[0]]
+          break if iterator.nil?
+          path.slice(0)
```

Comment:
and I think you want `slice!` not `slice`

Reply:
Yes! thanks to this comment, I have realized that I had a mistake with my test. Fixed the test, I am working on the solution